src/guides/v2.3/extension-dev-guide/intro/developers_roadmap.md

```diff
@@ -12,15 +12,92 @@ To satisfy the minimum required elements for creating or customizing your Magent
 
 *  [Declare component dependencies]({{ page.baseurl }}/extension-dev-guide/build/composer-integration.html) in `composer.json`.
 
+    ##### Example: `composer.json` file.
+    ```json
+    {
+        "name": "magento/module-config",
+        "description": "N/A",
+        "config": {
+            "sort-packages": true
+        },
+        "require": {
+            "php": "~7.1.3||~7.2.0",
+            "magento/framework": "102.0.*",
+            "magento/module-backend": "101.0.*",
+            "magento/module-cron": "100.3.*",
+            "magento/module-deploy": "100.3.*",
+            "magento/module-directory": "100.3.*",
+            "magento/module-email": "101.0.*",
+            "magento/module-media-storage": "100.3.*",
+            "magento/module-store": "101.0.*"
+        },
+        "type": "magento2-module",
+        "license": [
+            "OSL-3.0",
+            "AFL-3.0"
+        ],
+        "autoload": {
+            "files": [
+                "registration.php"
+            ],
+            "psr-4": {
+                "Magento\\Config\\": ""
+            }
+        },
+        "version": "101.1.2"
+    }
+    ```
+
    {:.bs-callout-tip}
    While you can manage dependencies on your own, it is a recommended and strongly encouraged best practice to use the `composer.json` file.
 
 *  [Register]({{ page.baseurl }}/extension-dev-guide/build/component-registration.html) the component using the `registration.php` file.
+
+    ##### Example: `registration.php` file.
+    ```php
+    <?php
+    use \Magento\Framework\Component\ComponentRegistrar;
+    ComponentRegistrar::register(ComponentRegistrar::MODULE, 'Magento_Config', __DIR__);
+    ```
+
 *  Use these component-specific XML definition files:
    *  Modules: [`module.xml`]({{ page.baseurl }}/extension-dev-guide/build/create_component.html)
+
+       ##### Example: `module.xml` file.
```

Comment:
I would suggest using relative path here to make it more clear where exactly this file should be placed/edited

```suggestion
       ##### Example: `etc/module.xml` file.
```

This suggestion also applies to all other examples.